Automated versioning pyproject

wolfiex opened the pull request:
### The pull requests ties the github tag to the code version. 

This should be accessible using the `packagename.version` . 

Additionally install requirements and package dependancies have been  correctly separated.

wolfiex committed:
automatic version linking to git tag

matthew-mizielinski commented:
in #64 I put the version number in data_request_api/__init__.py (so that a `pydoc data_request_api` shows this with limited effort)

wolfiex committed:
Update __init__.py

Import the version variables in the dynamically generated version.py occurring on library install.

wolfiex committed:
python -c "from data_request_api import *; print(version)"

wolfiex commented:
Good point @matthew-mizielinski ,I have now also added an import for  all the version variables from the dynamically generated `version.py`  to `__init__`. 

You can test it using `python -c "from data_request_api import *; print(version)"`

JamesAnstey committed:
Merge branch 'main' into versioning_pyproject
